src/github.ts

```diff
@@ -701,8 +707,12 @@ export class GitHub {
           message: graphCommit.message,
           files: [] as string[],
         };
-        if (graphCommit.associatedPullRequests.nodes.length > 0) {
-          const pullRequest = graphCommit.associatedPullRequests.nodes[0];
+        const pullRequest = graphCommit.associatedPullRequests.nodes.find(
+          pr => {
+            return pr.mergeCommit && pr.mergeCommit.oid === graphCommit.sha;
+          }
+        );
+        if (pullRequest) {
```

Comment:
makes sense though I'd be curious to understand better what a `pr.mergeCommit.oid` is in different cases of how the PR could have been merged (e.g. traditional "create a merge commit", vs "squash", vs "rebase"). I guess whatever the PR-Merge strategy, this logic will find the PR that literally is the reason for this `graphCommit.sha/pr.mergeCommit.oid` to exist in the history of the branch at all.

Reply:
Example responses:
* Rebase/merge [PR](https://github.com/chingor13/release-please-test/pull/6) [graphql](https://gist.github.com/chingor13/8bcdaf992020d89d0c404a3bbbbeff9a#file-gistfile1-json-L71-L154)
* Merge [PR](https://github.com/chingor13/release-please-test/pull/7) [graphql](https://gist.github.com/chingor13/8bcdaf992020d89d0c404a3bbbbeff9a#file-gistfile1-json-L8-L70)